src/Lachain.Core/Blockchain/SystemContracts/DeployContract.cs

```diff
@@ -29,6 +30,13 @@ public DeployContract(InvocationContext context)
 
         [ContractMethod(DeployInterface.MethodDeploy)]
         public ExecutionStatus Deploy(byte[] byteCode, SystemContractExecutionFrame frame)
+        {
+            if (frame.InvocationContext.Snapshot.Blocks.GetTotalBlockHeight() < HardforkHeights.Hardfork_2)
```

Comment:
this is probably inconsistent. 
We used GetTotalBlockHeight() > HardforkHeights.Hardfork_2 
in some places. This consistency may have some issues at the hardfork height.